### PR TITLE
hyperopt: init at 0.2.2

### PR DESCRIPTION
**Pull request by GuillaumeDesforges**
###### Motivation for this change
`hyperopt` was not part of nixpkgs.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Additional information
I have tested the package using the example from hyperopt readme.

### CODE REVIEW

File: pkgs/development/python-modules/hyperopt/default.nix [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+      inherit pname version;
+      sha256 = "0mf4zligyhracyfx02ayrvdnxwxsbr912sfby167pgi3fy3vjyfb";
+    };
+
+    meta = with stdenv.lib; {
+      homepage = "https://github.com/hyperopt/hyperopt";
+      description = "Hyperopt is a Python library for serial and parallel optimization over awkward search spaces";
+      license = {
+        fullName = "Copyright (c) 2013, James Bergstra All rights reserved.";
+        url = "https://raw.githubusercontent.com/hyperopt/hyperopt/c903de979e50f844130dcc08107038f95e87a061/LICENSE.txt";
```
Comment (symphorien) [inline pkgs/development/python-modules/hyperopt/default.nix]: this looks like bsd 3 clauses

Comment (GuillaumeDesforges) [reply pkgs/development/python-modules/hyperopt/default.nix]: Thanks

**Comment by GuillaumeDesforges** [comment]
[This issue](https://github.com/hyperopt/hyperopt/issues/580) is now solved, however there is still a problem with the tests.

The full log is here: https://gist.github.com/GuillaumeDesforges/9aa019307f05d98065d7f4c6f85f4f21

The main issue seems to be that files that are not `.py` files were not copied for the test. 

```
FileNotFoundError: [Errno 2] No such file or directory: '/build/hyperopt-0.2.2/hyperopt/atpe_models/scaling_model.json'
```

How can I add them to the list of files to copy ?

**Review by doronbehar — APPROVED** [review]

**Comment by doronbehar** [comment]
Have you tried consulting upstream?

**Comment by jonringer** [comment]
if it's needed for normla usage, then adding a files makes sense.

Still unusual if needed though

**Comment by timokau** [comment]
The FileNotFound issue can be resolved by fetching from GitHub instead of pypy:

```nix
src = fetchFromGitHub {
  owner = "hyperopt";
  repo = "hyperopt";
  rev = version;
  sha256 = "1x9nasjlck408f86bvb23vy31py1rjcy9bjqicy30bclrdbmh6x9";
};
```

which is weird since it seems like they should be distributing that file over pypi. That gives the next test failure however, which is due to a newer numpy version and some numerical instability:

https://github.com/hyperopt/hyperopt/issues/315

Other relevant issues:
- https://github.com/hyperopt/hyperopt/issues/354
- https://github.com/hyperopt/hyperopt/issues/350

And unfortunately this is not meregable with the networkx pin, since we cannot use multiple versions of networks in parallel.

**Review by timokau — CHANGES_REQUESTED** [review]
See comment.

**Comment by jonringer** [comment]
> which is weird since it seems like they should be distributing that file over pypi. That gives the next test failure however, which is due to a newer numpy version and some numerical instability:

I believe they would have to list the file(s) in their MANIFEST.in or list it as a data_dir in their `setup.py`. 

A lot of packages just assume you're going to pull down their wheel and blindly install what you get from pypi.

**Comment by GuillaumeDesforges** [comment]
> I believe they would have to list the file(s) in their MANIFEST.in or list it as a data_dir in their `setup.py`.
> 
> A lot of packages just assume you're going to pull down their wheel and blindly install what you get from pypi.

I think that this line is supposed to do just that and add all the data automatically:
https://github.com/hyperopt/hyperopt/blob/b593d18ada68abe96e84b43055dec0cc799a5891/setup.py#L98

**Comment by jonringer** [comment]
That's just to refactor out the logic, the important part is passing it to setup() https://github.com/hyperopt/hyperopt/blob/b593d18ada68abe96e84b43055dec0cc799a5891/setup.py#L146

either way, both parts are necessary, and the data should be getting exported if it's able to correctly find the data

**Comment by GuillaumeDesforges** [comment]
Deleted forked repo, recreated it, so I needed to create a new PR https://github.com/NixOS/nixpkgs/pull/82144

Sorry for the inconvenience.

**Pull request closed without merging**